File: cinder/Entrainment/src/ViewFilmGrain.cpp

//
//  ViewFilmGrain.cpp
//  Entrainment03
//
//  Created by Yi-Wen Lin on 2015/7/28.
//
//

#include "ViewFilmGrain.h"

#include "MeshUtils.h"

using namespace ci;
using namespace ci::app;
using namespace std;

ViewFilmGrain::ViewFilmGrain() : ViewCopy("shaders/copy.vert", "shaders/filmGrain.frag") {
}

ViewFilmGrain::ViewFilmGrain(string vsPath, string fsPath) : ViewCopy(vsPath, fsPath) {
}


void ViewFilmGrain::render(gl::TextureRef texture) {
    render(*texture);
}

void ViewFilmGrain::render(gl::Texture texture) {
    shader->bind();
    shader->uniform("texture", 0);
    shader->uniform("time", _count);
    shader->uniform("width", float(getWindowWidth()));
    shader->uniform("height", float(getWindowHeight()));
    texture.bind();
    gl::draw(mesh);
    texture.unbind();
    shader->unbind();
    
    _count += .01;
}

File: webgl/src/js/ViewFilmGrain.js
// ViewFilmGrain.js

var GL = bongiovi.GL;
var gl;
var glslify = require("glslify");

function ViewFilmGrain() {
	this.count = 0;
	bongiovi.ViewCopy.call(this, null, glslify("../shaders/panorama.frag"));
}

var p = ViewFilmGrain.prototype = new bongiovi.ViewCopy();
p.constructor = ViewFilmGrain;


p.render = function(texture) {
	if(!this.shader.isReady() ) return;

	this.shader.bind();
	this.shader.uniform("texture", "uniform1i", 0);
	this.shader.uniform("time", "uniform1f", this.count);
	this.count += .01;
	texture.bind(0);
	GL.draw(this.mesh);
};

module.exports = ViewFilmGrain;